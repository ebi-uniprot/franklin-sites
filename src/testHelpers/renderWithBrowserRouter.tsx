import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

// Inspired by https://testing-library.com/docs/example-react-router/

const renderWithBrowserRouter = (ui: ReactElement, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
};

export default renderWithBrowserRouter;
