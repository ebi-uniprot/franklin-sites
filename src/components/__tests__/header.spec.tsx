import { Link } from 'react-router-dom';
import {
  BasketIcon,
  Button,
  EnvelopeIcon,
  ExternalLink,
  HelpIcon,
  MainSearch,
  Dropdown,
} from '..';

import renderWithBrowserRouter from '../../testHelpers/renderWithBrowserRouter';

import Header from '../header';

describe('Header component', () => {
  it('should render', () => {
    const { asFragment } = renderWithBrowserRouter(
      <Header
        secondaryItems={
          <>
            <Link to="/">
              <HelpIcon />
            </Link>
            <Link to="/">
              <EnvelopeIcon />
            </Link>
            <Link to="/">
              <BasketIcon />
            </Link>
          </>
        }
        search={<MainSearch onTextChange={jest.fn()} onSubmit={jest.fn()} />}
      >
        <Link to="/here">here</Link>
        <Link to="/there">there</Link>
        <ExternalLink url="www.ebi.ac.uk">EBI</ExternalLink>
        <Button onClick={jest.fn()}>Action</Button>
        <Dropdown
          visibleElement={<Button variant="tertiary">sub links</Button>}
        >
          <Link to="/sub-link-a">sub link A</Link>
          <Link to="/sub-link-b">
            <span>sub link B</span>
          </Link>
        </Dropdown>
      </Header>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
