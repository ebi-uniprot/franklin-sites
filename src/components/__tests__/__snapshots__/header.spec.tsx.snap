// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="header__logo"
    >
      <a
        href="/"
      />
    </div>
    <div
      class="header__navigation"
    >
      <a
        href="/here"
      >
        here
      </a>
      <a
        href="/there"
      >
        there
      </a>
      <a
        class="external-link"
        href="www.ebi.ac.uk"
        rel="noopener"
        target="_blank"
      >
        EBI
        <test-file-stub
          data-testid="external-link-icon"
          width="12.5"
        />
      </a>
      <button
        class="button primary"
        type="button"
      >
        Action
      </button>
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="dropdown"
      >
        <button
          class="button tertiary"
          type="button"
        >
          sub links
        </button>
      </div>
    </div>
    <div
      class="header__search"
    >
      <form
        aria-label="Main search"
        class="main-search"
      >
        <div
          class="main-search__input-container"
        >
          <input
            aria-label="Text query"
            type="search"
            value=""
          />
        </div>
        <button
          class="button primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="header__secondary"
    >
      <div
        class="header__navigation"
      >
        <a
          href="/"
        >
          <test-file-stub />
        </a>
        <a
          href="/"
        >
          <test-file-stub />
        </a>
        <a
          href="/"
        >
          <test-file-stub />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
